### PR TITLE
community: load pdf from bytes using PDFLoaders

### PR DESCRIPTION
**Pull request by AyushModi123**
# **Description**
Added load pdf from bytes support in PyPDFLoader, PyPDFium2Loader, PDFMinerLoader, PyMuPDFLoader,  PDFPlumberLoader.

# **Dependencies**
None

# **Issue**
 This change fixes the issue #6265
 
# **Twitter handle**
@AyushModi426631

### CODE REVIEW

**Comment by majinmarco** [comment]
The lack of this feature to use with the GoogleDriveLoader was today's pain in the butt for me and it was fixed a day ago bruh 😭

**Comment by AyushModi123** [comment]
Hey @eyurtsev, can you please review this pull request..

**Comment by AyushModi123** [comment]
Hey @maximeperrindev,  can you please review this pull request.
It resolves an important issue which many developers face.

File: libs/community/langchain_community/document_loaders/pdf.py [hunk 1]
```diff
@@ -80,7 +80,9 @@ class BasePDFLoader(BaseLoader, ABC):
         clean up the temporary file after completion.
     """
 
-    def __init__(self, file_path: str, *, headers: Optional[Dict] = None):
+    def __init__(
+        self, file_path: Union[str, BytesIO], *, headers: Optional[Dict] = None
```
Comment (maximeperrindev) [inline libs/community/langchain_community/document_loaders/pdf.py:84]: I think talking about a `file_path`while manipulating bytes (which is a `file`then) is a bit confusing.

**Review by maximeperrindev — COMMENTED** [review]
Hi @AyushModi123 ! 
The PR looks good to me. Just a confusing variable name that could be changed.
Also, as i'm not a maintainer, I can't approve/merge it. Thanks for your work though

File: libs/community/langchain_community/document_loaders/pdf.py (hunk 1 above)
Comment (AyushModi123) [reply libs/community/langchain_community/document_loaders/pdf.py:84]: Thanks for reviewing the PR. I'll tweak the variable name. Appreciate your input!

**Pull request closed without merging**